### DISCUSSION TITLE
Image Cropper Overlaps Rotated Image Boundaries in Fabric.js

### DISCUSSION THREAD

**Discussion by osamakhn97** [fabricjs/fabric.js#9136] [Q&A]
In this issue, I am seeking guidance and assistance with a custom image cropper implementation using Fabric.js. The image cropper, implemented as an extension of the Fabric.js image object, functions flawlessly at zero degrees rotation. However, when the image is rotated to any non-zero degree, the cropping boundaries overlap with the image itself, leading to undesired results. I suspect that the problem lies in how coordinate transformations are handled during the rotation process, but I am unable to pinpoint the exact cause. I am looking for insights, suggestions, and code examples to help me properly restrict the cropping boundaries while the image is rotated. Any assistance from the community would be highly appreciated. Thank you for your time and expertise in resolving this issue.
Cropping is working fine, infact every functionality is working fine only I need to restrict the boundaries so that cropeer must not get out of the image boundaries. 
 
![i1](https://github.com/fabricjs/fabric.js/assets/95467997/ac739397-8740-44d4-948b-0ca8657325e1)

**Comment by lh-sn** [comment]
I have implemented cropping on V6. Solution could be  more complex that what you have right now. Can you share a reproduction for the issue you are facing?
![image](https://github.com/fabricjs/fabric.js/assets/137523466/df72ebe5-9133-439b-a79d-28f0f3d33d7a)

**Reply by ShaMan123** [reply]
We are planning to release cropping controls... 
@asturur

**Reply by lh-sn** [reply]
![image](https://github.com/fabricjs/fabric.js/assets/137523466/bf176d92-d4e1-4d60-ba90-b966379283d9)

@ShaMan123 how about clipping mask controls ?

**Reply by ShaMan123** [reply]
Can this be done by rendering another layer onto the image cache?

**Reply by ShaMan123** [reply]
Might be also possible using group... but more work I think

**Reply by lh-sn** [reply]
Rendering another layer onto the image cache works. About grouping, not sure. seems like it may not work

**Reply by ouyoungs** [reply]
> We are planning to release cropping controls... @asturur

is this released now ?

**Reply by Arkit-Sutariya** [reply]
Hello @lh-sn 
Could you kindly share the URL for your cropping demo created with V6?

**Comment by diegosystem32** [comment]
> In this issue, I am seeking guidance and assistance with a custom image cropper implementation using Fabric.js. The image cropper, implemented as an extension of the Fabric.js image object, functions flawlessly at zero degrees rotation. However, when the image is rotated to any non-zero degree, the cropping boundaries overlap with the image itself, leading to undesired results. I suspect that the problem lies in how coordinate transformations are handled during the rotation process, but I am unable to pinpoint the exact cause. I am looking for insights, suggestions, and code examples to help me properly restrict the cropping boundaries while the image is rotated. Any assistance from the community would be highly appreciated. Thank you for your time and expertise in resolving this issue. Cropping is working fine, infact every functionality is working fine only I need to restrict the boundaries so that cropeer must not get out of the image boundaries.
> 
> ![i1](https://private-user-images.githubusercontent.com/95467997/257144879-ac739397-8740-44d4-948b-0ca8657325e1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ3NDg5MDgsIm5iZiI6MTc0NDc0ODYwOCwicGF0aCI6Ii85NTQ2Nzk5Ny8yNTcxNDQ4NzktYWM3MzkzOTctODc0MC00NGQ0LTk0OGItMGNhODY1NzMyNWUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE1VDIwMjMyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThiY2FlZmEwZWQwY2YyYzIwMDQ3YTc2YTEyNWE5YzMzMmQ0YTY1YjViNzgwOGFhNTFlMWMwYmRlNGU0ZjU3MTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.OiSKCZXec5_N4WkTBKg8P5niuVC0P4Ho2cM04Jbp0jM)

Did you get it? Please share, I have the same case.

**Comment by zhe-he** [comment]
Why is there a need to limit the boundaries? I'm quite curious about its usage scenarios. This is a cropping demo, and I hope it will be helpful to you.

https://github.com/fabricjs/fabric.js/discussions/10487
